### PR TITLE
E2E Reset site kit

### PR DESCRIPTION
**Pull request by lucymtc**
## Summary
E2E Reset site kit

Addresses issue https://github.com/google/site-kit-wp/issues/275

## Checklist

- [x] My code is tested and passes existing unit tests.
- [x] My code has an appropriate set of unit tests which all pass.
- [x] My code is backward-compatible with WordPress 4.7 and PHP 5.4.
- [x] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [x] My code has proper inline documentation.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

File: tests/e2e/specs/plugin-reset.test.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
+/**
+ * WordPress dependencies
+ */
+import { deactivatePlugin, activatePlugin, visitAdminPage } from '@wordpress/e2e-test-utils';
+
+describe( 'Reset plugin', () => {
+	beforeAll( async() => {
+		await activatePlugin( 'e2e-tests-auth-plugin' );
+		await activatePlugin( 'google-site-kit' );
```
Comment (swissspidy) [inline tests/e2e/specs/plugin-reset.test.js]: The Site Kit plugin should actually be already active during E2E tests (done in the Docker setup). The only exception where it needs to be (de)activated is in the test for the plugin activation itself.

File: tests/e2e/specs/plugin-reset.test.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+		await deactivatePlugin( 'e2e-tests-auth-plugin' );
+		await deactivatePlugin( 'google-site-kit' );
+	} );
+
+	beforeEach( async() => {
+		await visitAdminPage( 'admin.php', 'page=googlesitekit-settings' );
+		await page.waitForSelector( '.mdc-tab-scroller__scroll-content' );
+
+		// Click on Admin Settings Tab.
+		await page.$$eval( '.mdc-tab-scroller__scroll-content button', tabs => {
```
Comment (swissspidy) [inline tests/e2e/specs/plugin-reset.test.js]: A bit simplified and easier to read:

```js
await expect( page ).toClick( '.mdc-tab-scroller__scroll-content button', { text: 'Admin Settings' } );
```

File: tests/e2e/specs/plugin-reset.test.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+			Array.apply( null, tabs )
+				.find( element => 'Admin Settings' === element.textContent )
+				.click();
+		} );
+
+		// Click on Reset Site Kit.
+		await page.evaluate( () => {
```
Comment (swissspidy) [inline tests/e2e/specs/plugin-reset.test.js]: Likewise:

```js
await expect( page ).toClick( '.googlesitekit-cta-link', { text: 'Reset Site Kit' } );
```

Although I'd argue that clicking on the reset button should be part of each test to make it clear that the step needs confirmation.

File: tests/e2e/specs/plugin-reset.test.js [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+			document.evaluate(
+				'//button[contains(text(), "Reset Site Kit")]', document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null
+			).singleNodeValue.click();
+		} );
+	} );
+
+	it( 'On Reset Site Kit click, confirm dialog displays', async() => {
+		await page.waitForSelector( '.mdc-dialog[aria-hidden="false"]' );
+
+		const isHidden = await page.$eval( '.mdc-dialog[aria-hidden="false"]', dialog => {
```
Comment (swissspidy) [inline tests/e2e/specs/plugin-reset.test.js]: Suggestion:

```js
await expect( page ).toMatchElement( '.mdc-dialog.mdc-dialog--open .mdc-dialog__cancel-button', { text: 'Cancel' } );
```

This only matches the cancel button of the opened dialog.

File: tests/e2e/specs/plugin-reset.test.js [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+				return button && 'Reset' === button.textContent;
+			} );
+			if ( dialog ) {
+				return true;
+			}
+
+			return false;
+		} );
+
+		expect( isHidden ).toBe( true );
```
Comment (swissspidy) [inline tests/e2e/specs/plugin-reset.test.js]: The whole test could be simplified to something like this:

```js
await expect( page ).toMatchElement( '.mdc-dialog.mdc-dialog--open' );
await expect( page ).toClick( '.mdc-dialog.mdc-dialog--open .mdc-dialog__cancel-button', { text: 'Cancel' } );
await expect( page ).not.toMatchElement( '.mdc-dialog.mdc-dialog--open' );
```

File: tests/e2e/specs/plugin-reset.test.js [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+	} );
+
+	it( 'Reset dialog button disconnects site kit', async() => {
+
+		// Click Reset button.
+		await page.$$eval( '.mdc-dialog__actions button', buttons => {
```
Comment (swissspidy) [inline tests/e2e/specs/plugin-reset.test.js]: Again, simplified:

```js
await expect( page ).toClick( '.mdc-dialog.mdc-dialog--open .mdc-dialog__cancel-button', { text: 'Cancel' } );
```

**Comment by swissspidy** [comment]
@aaemnnosttv Can you reply to googlebot when you get a chance? 🙂

**Comment by aaemnnosttv** [comment]
@googlebot I consent.

File: tests/e2e/specs/plugin-reset.test.js [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+describe( 'Reset plugin', () => {
+	beforeAll( async() => {
+		await activatePlugin( 'e2e-tests-auth-plugin' );
+		await activatePlugin( 'google-site-kit' );
+	} );
+
+	afterAll( async() => {
+		await deactivatePlugin( 'e2e-tests-auth-plugin' );
+		await deactivatePlugin( 'google-site-kit' );
```
Comment (swissspidy) [inline tests/e2e/specs/plugin-reset.test.js]: As mentioned above, no need to deactivate the plugin.

File: tests/e2e/specs/plugin-reset.test.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+		await expect( page ).toMatchElement( '.googlesitekit-page-header__title', { text: 'Settings' } );
+
+		await page.waitForSelector( 'button.mdc-tab' );
+
+		// Click on Admin Settings Tab.
+		await expect( page ).toClick( 'button.mdc-tab', { text: 'Admin Settings' } );
+		await page.waitForSelector( '.googlesitekit-settings-module__footer' );
+	} );
+
+	it( 'On Reset Site Kit click, confirm dialog displays.', async() => {
```
Comment (swissspidy) [inline tests/e2e/specs/plugin-reset.test.js]: Test names should start with a lowercase letter, makes for nice readable and consistent reports 🙂 

Think of it as a sentence: `it ...`, e.g. `it should display a confirmation dialog`, `it should hide the confirmation dialog on cancel`, `it should disconnect Site Kit`

Comment (aaemnnosttv) [reply tests/e2e/specs/plugin-reset.test.js]: Updated 👍

Comment (swissspidy) [reply tests/e2e/specs/plugin-reset.test.js]: Excellent, thanks!

**Pull request merged**